store/src/commonMain/kotlin/com/dropbox/external/store5/impl/MemoryLruCache.kt

```diff
@@ -0,0 +1,128 @@
+@file:Suppress("UNCHECKED_CAST")
+
+package com.dropbox.external.store5.impl
+
+import com.dropbox.external.store5.Persister
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.sync.Mutex
+
+/**
+ * Thread-safe LRU cache implementation.
+ */
+class MemoryLruCache(private val maxSize: Int) : Persister<String> {
+    internal var cache = mutableMapOf<String, Node<*>>()
+    internal var head = headPointer
+    internal var tail = tailPointer
+
+    private val lock = Mutex()
+
+    init {
+        head.next = tail
+        tail.prev = head
+    }
+
+    override fun <Output : Any> read(key: String): Flow<Output?> = flow {
+        lock.lock()
```

Comment:
You could also do `lock.withLock { }`, which has the added benefit of unlocking the mutex if the lambda block throws.

Reply:
https://github.com/MobileNativeFoundation/Store/pull/450/commits/48bdb1c60797438ff638d3cd3541158eead88939